File: 2533-bitwise-xor-of-all-pairings/2533-bitwise-xor-of-all-pairings.java
class Solution {
    public int xorAllNums(int[] nums1, int[] nums2) {
        int result = 0;
        int n1 = nums1.length;
        int n2 = nums2.length;
        
        // If nums2.length is odd, each number in nums1 appears odd times in final result
        if ((n2 & 1) == 1) {
            for (int num : nums1) {
                result ^= num;
            }
        }
        
        // If nums1.length is odd, each number in nums2 appears odd times in final result
        if ((n1 & 1) == 1) {
            for (int num : nums2) {
                result ^= num;
            }
        }
        
        return result;
    }
}

File: 2533-bitwise-xor-of-all-pairings/2533-bitwise-xor-of-all-pairings.c
int xorAllNums(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int result = 0;
    
    // If nums2Size is odd, each number in nums1 appears odd times in final result
    if (nums2Size & 1) {
        for (int i = 0; i < nums1Size; i++) {
            result ^= nums1[i];
        }
    }
    
    // If nums1Size is odd, each number in nums2 appears odd times in final result
    if (nums1Size & 1) {
        for (int i = 0; i < nums2Size; i++) {
            result ^= nums2[i];
        }
    }
    
    return result;
}